pkg/virt-controller/watch/vmi.go

```diff
@@ -1340,9 +1340,25 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		}
 
 		if vmiSpecIfaces, vmiSpecNets, dynamicIfacesExist := network.CalculateInterfacesAndNetworksForMultusAnnotationUpdate(vmi); dynamicIfacesExist {
-			if err := c.updateMultusAnnotation(vmi.Namespace, vmiSpecIfaces, vmiSpecNets, pod); err != nil {
+			errMessage := fmt.Errorf("failed to hot{un}plug network interfaces for vmi [%s/%s]", vmi.GetNamespace(), vmi.GetName())
+
+			var networkToIPAMClaimParams map[string]libipam.IPAMClaimParams
+			if c.clusterConfig.PersistentIPsEnabled() {
+				var err error
+				networkToIPAMClaimParams, err = c.ipamClaimsManager.GetNetworkToIPAMClaimParams(
+					vmi.Namespace,
+					vmi.Name,
+					vmispec.FilterMultusNonDefaultNetworks(vmiSpecNets))
+				if err != nil {
+					return &syncErrorImpl{
+						err:    fmt.Errorf("%s: %w", errMessage, err),
+						reason: FailedHotplugSyncReason,
```

Comment:
@oshoval It looks like the following code:
```
				networkToIPAMClaimParams, err = c.ipamClaimsManager.GetNetworkToIPAMClaimParams(
					vmi.Namespace,
					vmi.Name,
					vmispec.FilterMultusNonDefaultNetworks(vmiSpecNets))
				if err != nil {
```
can be moved inside the `updateMultusAnnotations` function

Reply:
There were few reasons for having it here,
Having the FG check highest level possible
Allowing passing the ipamClaimParams to the function, instead having it reading it from NADs when unit testing
(and then we would also need to toggle FG when unit testing it)
if still desired i will it inside, we just had few iterations on this / some stuff alike so i prefer to see if we want to